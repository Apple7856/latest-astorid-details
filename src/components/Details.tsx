import { Container, Typography } from '@mui/material'
import React from 'react'
import { useLocation } from 'react-router-dom'

const Details = () => {
    const location:any = useLocation();
    return (
        <Container maxWidth="md" sx={{ backgroundColor: "#b3ffb3", width: "100%", height: "100vh", display: "flex", flexDirection: "column", alignItems: "center", justifyContent: "center" }}>
            <Typography variant='h3' data-testid="heading">Astroid Details</Typography>
            <Typography data-testid="nameTest">name: {location.state.name}</Typography>
            <Typography data-testid="nasa_jpl_url">nasa_jpl_url: {location.state.nasa_jpl_url}</Typography>
            <Typography data-testid="is_potentially_hazardous_asteroid">Is_potentially_hazardous_asteroid: {location.state.is_potentially_hazardous_asteroid}</Typography>
        </Container>
    )
}

export default Details