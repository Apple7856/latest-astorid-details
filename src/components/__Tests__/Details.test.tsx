import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Details from '../Details';

beforeEach(() => {
    document.body.innerHTML = "";
});

afterEach(() => {
    cleanup();
});

const mockUseLocationValue = {
    pathname: "/details",
    search: '',
    hash: '',
    state: {
        name: "Harry",
        nasa_jpl_url: "xyz",
        is_potentially_hazardous_asteroid: "true"
    }
}

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom") as {},
    useLocation: jest.fn().mockImplementation(() => {
        return mockUseLocationValue;
    })
}));


describe("Details Component Test", () => {
    
    test("testing", () => {
        expect(true).toBe(true);
    })

    test("Input Field Test", () => {
        const component = render(<Details />);
        const heading = component.getByTestId('heading');
        expect(heading.innerHTML).toBe('Astroid Details');
    })
    test("Name Text Test", () => {
        const component = render(<Details />);
        const nameTest = component.getByTestId('nameTest');
        expect(nameTest.innerHTML).toBe("name: "+mockUseLocationValue.state.name);
    })
    test("nasa_jpl_url Text Test", () => {
        const component = render(<Details />);
        const nasa_jpl_url = component.getByTestId('nasa_jpl_url');
        expect(nasa_jpl_url.innerHTML).toBe("nasa_jpl_url: "+mockUseLocationValue.state.nasa_jpl_url);
    })
    test("Is_potentially_hazardous_asteroid Text Test", () => {
        const component = render(<Details />);
        const is_potentially_hazardous_asteroid = component.getByTestId('is_potentially_hazardous_asteroid');
        expect(is_potentially_hazardous_asteroid.innerHTML).toBe("Is_potentially_hazardous_asteroid: "+mockUseLocationValue.state.is_potentially_hazardous_asteroid);
    })
    test("Create SnapShot", () => {
        const tree = renderer.create(<Details />);
        expect(tree).toMatchSnapshot();
    })

})