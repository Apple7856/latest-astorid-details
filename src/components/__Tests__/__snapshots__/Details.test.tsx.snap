// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component Test Create SnapShot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthMd css-1n2gf07-MuiContainer-root"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    data-testid="heading"
  >
    Astroid Details
  </h3>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    data-testid="nameTest"
  >
    name: 
    Harry
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    data-testid="nasa_jpl_url"
  >
    nasa_jpl_url: 
    xyz
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    data-testid="is_potentially_hazardous_asteroid"
  >
    Is_potentially_hazardous_asteroid: 
    true
  </p>
</div>
`;
